### PR TITLE
Correctly create cloud account from mobile

### PR DESCRIPTION
**Pull request by janmulder**
The creation of a cloud account from mobile was broken. This fixes it. Basically, we need to go online for a moment, and setup a correct local and remote repo for the cloud storage.

Tested for the following scenarios: 1) inital account creation including PIN handling from mobile, from a clean install .
2) open an already validated cloud account from a clean install.
3) open no-cloud style local account.
4) Switch between 2 already validated could accounts.
5) Try to create a cloud account without data connection.

Notice that scenario 4) does not work perfectly. A restart of the app is needed to see the new logbook. So that is to be fixed.

Scenario 5) seems a non realistic corner case. This does not work in a graceful way. The user needs to remove the app, install it again, and retry with data connection.

Further notice this is backgroud/core processing only. So no QML UI
changes as proposed (for example) bij Davide.

See the individual commits for more details.

Signed-off-by: Jan Mulder <jlmulder@xs4all.nl>

### CODE REVIEW

**Comment by janmulder** [comment]
Travis error: CMake Error at Qt/5.9/gcc_64/lib/cmake/Qt5/Qt5Config.cmake:28 (find_package):
Could not find a package configuration file provided by "Qt5WebKitWidgets"

Hmm. Something QtLoation/marble related?

**Comment by dirkhh** [comment]
I must have messed something up with my build.sh changes. We should not be asking for WebKit.
I'll look at it.

**Comment by dirkhh** [comment]
My build.sh changes weren't entirely broken - what was broken was the hack in .travis.yml to turn of WebKit. Fixed now

**Comment by mturkia** [comment]
On Sat, Jul 29, 2017 at 1:18 AM, janmulder <notifications@github.com> wrote:

> The creation of a cloud account from mobile was broken. This fixes it.
> Basically, we need to go online for a moment, and setup a correct local and
> remote repo for the cloud storage.
>
> Tested for the following scenarios: 1) inital account creation including
> PIN handling from mobile, from a clean install .
> 2) open an already validated cloud account from a clean install.
> 3) open no-cloud style local account.
> 4) Switch between 2 already validated could accounts.
> 5) Try to create a cloud account without data connection.
>
> Notice that scenario 4) does not work perfectly. A restart of the app is
> needed to see the new logbook. So that is to be fixed.
>
> Scenario 5) seems a non realistic corner case. This does not work in a
> graceful way. The user needs to remove the app, install it again, and retry
> with data connection
>
This actually does sound like a somewhat feasible scenario to me as we do
frequent bad connectivity locations for diving. And I can see someone
promoting our software in such a location. Download of the app might
succeed eventually while poor wifi will cause the init of cloud storage to
fail.

miika

**Comment by janmulder** [comment]
@mturkia : yes, I do understand your rationale. I do not like the behavior of scenario 5) either. However, when on a location with poor mobile data (with mobile data I mean, any data link, being wifi or any style mobile provider data link), it is also not realistic that you can receive the app from the play store to begin with, and receive the PIN email from the server, and when you can, you also have mobile data to do a successful cloud setup.  Further, my strategy in those situations (demo at a location without any mobile data, but having the app on the device) would be to use the no-cloud option. This can be transferred later to a cloud account. This is something I did not test (as I did see that code, and did not touch any of it).

So, yes, I could imagine the use case of trying to setup a cloud account without mobile data, that gets (with a toast message) diverted to a no cloud status, but that would probably confuse regular users as well.

Basically, I consider the only valid solution to this case: a transparent and error free setup of a cloud account in any situation (so also without mobile data). That gets "repaired" at the first restart with proper mobile data. While this seems simple, I expect numerous corner cases.

Let me finish with: patches are welcome :-)
